papers/scipy/code/signal/opt_lowpass.py

```diff
@@ -0,0 +1,153 @@
+
+import numpy as np
+from scipy.fftpack import next_fast_len
+from scipy.signal import remez, freqz
+import matplotlib.pyplot as plt
+
+
+def bellanger_estimate(deltap, deltas, width, fs):
+    """
+    Estimate the number of taps required for the given filter specifications.
+    """
+    n = (-2/3)*np.log10(10*deltap*deltas)*fs/width
+    n = int(np.ceil(n))
+    return n
+
+
+def remez_lowpass(deltap, deltas, cutoff, width, fs):
+
+    numtaps = bellanger_estimate(deltap, deltas, width, fs)
+    numtaps |= 1  # Ensure an odd number of taps.
+    trans_lo = cutoff - 0.5*width
+    trans_hi = cutoff + 0.5*width
+    taps = remez(numtaps,
+                 bands=[0, trans_lo, trans_hi, 0.5*fs],
+                 desired=[1, 0],
+                 weight=[1/deltap, 1/deltas],
+                 Hz=fs)
+    return taps
+
+
+#---------------------------------------
+# User inputs...
+
+# Frequency values in Hz
+fs = 1000.0
+cutoff = 180.0
+width = 30.0
+# Desired pass band ripple and stop band attenuation
+deltap = 0.005
+deltas = 0.002
+
+print("Pass band:  1 ± %g ([%.3g, %.3g] dB)" %
```

Comment:
Fails to run as well:

    SyntaxError: Non-ASCII character '\xc2' in file opt_lowpass.py on line 42, but no encoding declared